File: src/main/intellibitz/sted/ui/AboutSTED.java
package sted.ui;

import sted.io.Resources;
import sted.widgets.AboutDialog;

/**
 * Displays the About message.
 */
public class AboutSTED
        extends AboutDialog {

    private static AboutSTED aboutSTED;

    public static synchronized AboutSTED getInstance() {
        if (aboutSTED == null) {
            aboutSTED = new AboutSTED();
        }
        return aboutSTED;
    }

    private AboutSTED() {
        super(Resources.getResource(Resources.TITLE_ABOUT_STED),
                AboutText.getInstance());
    }

}

File: src/main/intellibitz/sted/ui/AboutSTED.kt
package sted.ui

import sted.io.Resources
import sted.widgets.AboutDialog
import java.awt.Component

class AboutSTED(title: String?, aboutDescriptor: Component?) : AboutDialog(title, aboutDescriptor) {
    companion object {
        val instance: AboutSTED by lazy {
            AboutSTED(
                Resources.getResource("title.about.sted"),
                AboutText.instance
            )
        }
    }
}